app/components/ui/molecules/PeoplePicker.js

```diff
@@ -145,11 +146,71 @@ class PeoplePicker extends React.Component {
     this.setState({ searchValue })
   }
 
-  filterPeople(people) {
-    const searchValue = this.state.searchValue.toLowerCase()
-    return people.filter(person =>
-      person.name.toLowerCase().includes(searchValue),
-    )
+  filterPeople = (people, searchValue, field) => {
+    if (!searchValue) return people
+
+    const inputValue = searchValue.trim().toLowerCase()
+    if (!inputValue) return people
+
+    if (inputValue.split(' ').length > 1) {
+      const matches = people.filter(person =>
+        person[field].toLowerCase().includes(inputValue),
+      )
+      return matches.map(elem => {
+        const clonedElem = cloneDeep(elem)
+        clonedElem.matchIndex = elem[field].toLowerCase().indexOf(inputValue)
+        return clonedElem
+      })
+    }
+    return this.filterPeopleSingle(people, inputValue, field)
+  }
+
+  filterPeopleSingle = (people, inputValue, field) => {
+    let maxWords = 0
+    people.forEach(person => {
+      const numWords = person[field].match(/[^ ]+/g).length
+      if (numWords > maxWords) {
+        maxWords = numWords
+      }
+    })
+
+    let j
+    let matches = []
+    const matchIndex = []
+    const re = new RegExp(`^${inputValue}`)
+
+    const filterNthMatches = (person, n) => {
+      const words = person[field].match(/[^ ]+/g)
+      if (words.length <= n) {
+        return false
+      }
+      const match = re.exec(words[n].toLowerCase())
+      if (match) {
+        matchIndex[j] = 0
+        for (let k = 0; k < n; k += 1) {
+          matchIndex[j] += words[k].length + 1
+        }
+        j += 1
+        return true
+      }
+      return false
+    }
+
+    const addMatchIndex = person => {
+      const clonedPerson = cloneDeep(person)
+      clonedPerson.matchIndex = matchIndex[j]
+      j += 1
+      return clonedPerson
+    }
+
+    for (let i = 0; i < maxWords; i += 1) {
+      /* matches for ith word */
+      j = 0
```

Comment:
Maybe you could explain how `j` is used? Not sure I understand, not sure its entirely clear... could we encapsulate this in some way?